lib/sr.py

```diff
@@ -166,7 +166,7 @@ def get_type(self) -> str:
             self._type = self.pool.master.xe("sr-param-get", {"uuid": self.uuid, "param-name": "type"})
         return self._type
 
-    def create_vdi(self, name_label: str, virtual_size: int = 64, image_format: Optional[str] = None) -> VDI:
+    def create_vdi(self, name_label: str, virtual_size: int | str = 64, image_format: Optional[str] = None) -> VDI:
```

Comment:
I would prefer to choose between int or string. String make sense with how you want to use it `1GiB`, so we could replace the default value to be a string and check caller to change them to use a string.